strawberryfields/apps/graph/dense.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+r"""
+Dense subgraph identification
+=============================
+
+**Module name:** :mod:`strawberryfields.future.apps.graph.dense`
+
+.. currentmodule:: strawberryfields.future.apps.graph.dense
+
+The frontend module for users to find dense subgraphs. The :func:`find_dense` function
+provides approximate solutions to the densest-:math:`k` subgraph problem
+:cite:`arrazola2018using`, which is NP-hard. This problem considers an undirected graph :math:`G
+= (V, E)` of :math:`N` nodes :math:`V` and a list of edges :math:`E`, and sets the objective of
+finding a :math:`k`-vertex subgraph with the greatest density. In this setting, subgraphs
+:math:`G[S]` are defined by nodes :math:`S \subset V` and corresponding edges :math:`E_{S}
+\subseteq E` that have both endpoints in :math:`S`. The density of a subgraph is given by
+
+.. math:: d(G[S]) = \frac{2 |E_{S}|}{|S|(|S|-1)},
+
+where :math:`|\cdot|` denotes cardinality, and the densest-:math:`k` subgraph problem can be
+written succinctly as
+
+.. math:: {\rm argmax}_{S \in \mathcal{S}_{k}} d(G[S])
+
+with :math:`\mathcal{S}_{k}` the set of all possible :math:`k` node subgraphs. This problem grows
+combinatorially with :math:`{N \choose k}` and is NP-hard in the worst case.
+
+The :func:`find_dense` function provides access to heuristic algorithms for finding
+approximate solutions. At present, random search is the heuristic algorithm provided, accessible
+through the :func:`random_search` function. This algorithm proceeds by randomly generating a set
+of :math:`k` vertex subgraphs and selecting the densest. Sampling of subgraphs can be achieved
+both uniformly at random and also with a quantum sampler programmed to be biased toward
+outputting dense subgraphs.
+
+Summary
+-------
+
+.. autosummary::
+    METHOD_DICT
+    OPTIONS_DEFAULTS
+    find_dense
+    random_search
+
+Code details
+------------
```

Comment:
This wills stop Code Details from showing up in the sidebar TOC (something we are in the process of removing from SF documentation)
```suggestion
^^^^^^^^^^^^
```